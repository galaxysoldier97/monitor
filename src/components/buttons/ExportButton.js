import React, { useState } from 'react';
import { IconButton, Snackbar, Tooltip } from '@material-ui/core';
import { t } from 'mt-react-library/functions';
import GetAppIcon from '@material-ui/icons/GetApp';
import TecrepSimCardService from '../../services/equipments/TecrepSimCardService';
import { getErrorMessage } from '../../helpers/fetchHelper';
import CircularProgress from '@material-ui/core/CircularProgress';
import TecrepCpeService from '../../services/equipments/TecrepCpeService';
import TecrepAncillaryEquipmentsService from '../../services/equipments/TecrepAncillaryEquipmentsService';
import { getCurrentDate, downloadExcelContent } from '../../helpers/exportHelper';
import PropTypes from "prop-types";

export const exportableEntity = Object.freeze({
  simcards: 'simcards',
  cpe: 'cpe',
  ancillaryEquipments: 'ancillaryEquipments'
});

const getExportFunction = (entity, filters) => {
  switch (entity) {
    case exportableEntity.simcards:
      return TecrepSimCardService.exportSimCards(filters);
    case exportableEntity.cpe:
      return TecrepCpeService.exportCpe(filters);
    case exportableEntity.ancillaryEquipments:
      return TecrepAncillaryEquipmentsService.exportAncillaryEquipments(filters);
  }
};
export const ExportButton = ({entity, filters}) => {
  const [notification, setNotification] = useState({visible: false, message: ''});
  const [isDownloading, setIsDownloading] = useState(false);
  const download = () => {
    setIsDownloading(true);
    getExportFunction(entity, filters)
    .then(r => downloadExcelContent(r.data, `${entity}-${getCurrentDate().replaceAll(':','_')}`))
    .catch(err => setNotification({visible: true, message: getErrorMessage(err)}))
    .finally(() => setIsDownloading(false));
  };

  return (
    <React.Fragment>
      {isDownloading ? <CircularProgress color="primary" size={20} /> :
        <Tooltip title={t('export')} onClick={download}>
        <IconButton size="small"><GetAppIcon /></IconButton>
      </Tooltip>}
      <Snackbar
        open={notification.visible}
        message={notification.message}
        autoHideDuration={4000}
        onClose={() => setNotification({...notification, visible: false})}
      />
    </React.Fragment>
  );
};
ExportButton.propTypes = {
  filters: PropTypes.object,
  entity: PropTypes.object,
};
