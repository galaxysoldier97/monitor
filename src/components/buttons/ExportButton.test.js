import React from 'react';
import renderer from 'react-test-renderer';
import { exportableEntity, ExportButton } from './ExportButton';

test('ExportButton snapshot', () => {
  const component = renderer.create(<ExportButton entity={exportableEntity.simcards} filters={{name: 'peppa'}} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
