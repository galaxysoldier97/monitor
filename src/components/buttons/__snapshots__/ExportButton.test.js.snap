// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportButton snapshot 1`] = `
<button
  aria-describedby={null}
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title={null}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
